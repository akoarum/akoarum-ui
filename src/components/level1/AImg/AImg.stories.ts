import { storiesOf } from '@storybook/vue'
import { withKnobs, text } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import AImg from './AImg.vue'

storiesOf('level1/AImg', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { AImg },
    props: {
      src: { default: text('src', 'https://dummyimage.com/300x200/ccc/fff.png') },
      alt: { default: text('alt', '') },
      srcset: { default: text('srcset', '') },
    },
    template: `
      <AImg :src="src" :srcset="srcset" :alt="alt" />
    `
  }), { info: {} })
