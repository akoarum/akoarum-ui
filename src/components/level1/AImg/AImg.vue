<template>
  <img :src="src" :srcset="srcset" :alt="alt" class="aImg" />
</template>

<script lang="ts">
import { defineComponent } from '@vue/composition-api'

export default defineComponent({
  props: {
    src: { type: String, required: true },
    alt: { type: String, required: true },
    srcset: { type: String }
  }
})
</script>

<style lang="scss" scoped>
.aImg {
}
</style>
