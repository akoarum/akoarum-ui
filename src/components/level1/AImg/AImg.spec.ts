import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import AImg from './AImg.vue'

describe('AImg', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(AImg, {
      propsData: {
        src: 'https://dummyimage.com/300x200/ccc/fff.png',
        srcset: 'https://dummyimage.com/600x400/ccc/fff.png 2x, https://dummyimage.com/300x200/ccc/fff.png 1x',
        alt: '代替テキスト'
      },
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
