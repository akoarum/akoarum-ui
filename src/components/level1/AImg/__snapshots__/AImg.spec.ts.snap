// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AImg initialized snapshot 1`] = `
<img
  alt="代替テキスト"
  class="aImg"
  src="https://dummyimage.com/300x200/ccc/fff.png"
  srcset="https://dummyimage.com/600x400/ccc/fff.png 2x, https://dummyimage.com/300x200/ccc/fff.png 1x"
/>
`;
